app.py

```diff
@@ -72,21 +27,44 @@ def get_authorization_header(root, discharge):
     """
 
     bound = Macaroon.deserialize(root).prepare_for_request(
-        Macaroon.deserialize(discharge))
+        Macaroon.deserialize(discharge)
+    )
 
     return 'Macaroon root={}, discharge={}'.format(root, bound.serialize())
 
 
+def is_connected():
+    return (
+        'openid' in flask.session or
+        'macaroon_discharge' in flask.session or
+        'macaroon_root' in flask.session
+    )
```

Comment:
Here I considered that you are logged in if you have one of those things in your session. Normally I would only need the openid stuff and only require the macaroon when I require it